includes/classes/Authentication/Passwords.php

```diff
@@ -374,4 +384,44 @@ public function enforce_for_user( $user_id ) {
 
 		return $enforce;
 	}
+
+	/**
+	 * Check if password is secure by querying the Have I Been Pwned API.
+	 *
+	 * @param string $password Password to validate.
+	 *
+	 * @return bool True if password is ok, false if it shows up in a breach.
+	 */
+	protected function is_password_secure( $password ): bool {
+		$hash   = strtoupper( sha1( $password ) );
+		$prefix = substr( $hash, 0, 5 );
+		$suffix = substr( $hash, 5 );
+
+		$response = wp_remote_get( self::API_URL . $prefix );
+
+		// Allow for a failed request to the HIPB API.
+		if ( is_wp_error( $response ) ) {
+			return true;
+		}
```

Comment:
**question (non-blocking):** Remond me, does `wp_remote_get()` always return a `WP_Error` for non-200 responses? If not, should we also check `wp_remote_retrieve_response_code()`?